include/xtensor/xfile_array.hpp

```diff
@@ -577,7 +577,10 @@ namespace xt
             // read new file
             if (enable_io(path))
             {
+                // file format may not preserve shape
+                const auto s = m_storage.shape();
                 m_io_handler.read(m_storage, path);
+                m_storage.reshape(s);
```

Comment:
Many questions regarding this (and related to the question to move this to xtensor-io):
- How do we know that the current shape should be reused after reading the buffer?
- How do we handle initialization of a file_array_container (meaning we read for the first time)?

Reshaping in xtensor-io could solve these issues for formats that store the shape, but I don't know what we should do with other formats.

Reply:
> How do we know that the current shape should be reused after reading the buffer?

What other choice do we have?

> How do we handle initialization of a file_array_container (meaning we read for the first time)?

We initialize to 0:
https://github.com/xtensor-stack/xtensor/blob/d0b55c294882cadef3dcd3f81b2c9b5248a527e4/include/xtensor/xdisk_io_handler.hpp#L60

> Reshaping in xtensor-io could solve these issues for formats that store the shape, but I don't know what we should do with other formats.

We could also reshape for formats that don't store the shape, but we need to evaluate the expression to retrieve its shape:
https://github.com/xtensor-stack/xtensor-io/blob/31cc5f0c9e6ceb6af858a019d17e439aaf1ed85f/include/xtensor-io/xio_binary.hpp#L133